### PR TITLE
[#IOCOM-1036] Added UserRCConfig model to bind users to rc configurations

### PR DESCRIPTION
**Pull request by michaeldisaro**
#### List of Changes
Added UserRCConfig model

#### Motivation and Context
We need to bind users to rc configurations to avoid expensive cross partition queries during rc config list by userId.

#### Types of changes

- [x] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Review by Garma00 — APPROVED** [review]
lgtm apart of the linter and a little observation

File: src/models/user_rc_configuration.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import { NonEmptyString } from "@pagopa/ts-commons/lib/strings";
+import { Container } from "@azure/cosmos";
+import { CosmosResource, CosmosdbModel } from "../utils/cosmosdb_model";
+
+export const USER_RC_CONFIGURATIONS_COLLECTION_NAME = "user-configurations";
+const USER_RC_CONFIGURATIONS_MODEL_PK_FIELD = "userId";
+
+const UserRCConfiguration = t.interface({
+  userId: NonEmptyString,
+  id: NonEmptyString
```
[the remark this answers is not in the document]
Comment (Garma00) [reply src/models/user_rc_configuration.ts:11]: Ok it's fine `NonEmptyString` then 😄

File: src/models/user_rc_configuration.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+   * @param container the Cosmos container client
+   */
+  constructor(container: Container) {
+    super(
+      container,
+      UserRCConfiguration,
+      RetrievedUserRCConfiguration
+    );
+  }
+}
```
Comment (lucacavallaro) [inline src/models/user_rc_configuration.ts:44]: can't be removed?

File: src/models/user_rc_configuration.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  UserRCConfiguration,
+  AzureCosmosResource
+]);
+export type RetrievedUserRCConfiguration = t.TypeOf<
+  typeof RetrievedUserRCConfiguration
+>;
+
+/**
+ * @deprecated This model is deprecated, use the one inside ./remote_content.ts instead
+ */
```
Comment (Garma00) [inline src/models/user_rc_configuration.ts:25]: :p

File: src/models/user_rc_configuration.ts (hunk 2 above)
Comment (michaeldisaro) [reply src/models/user_rc_configuration.ts:44]: it's been removed in another dedicated and  just released PR.

**Review by Garma00 — APPROVED** [review]

**Pull request merged**